libchihua-msg/include/orchestrator/connection-server.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include <stdbool.h>
+#include <systemd/sd-event.h>
+
+#include "../common/list.h"
+#include "../common/memory.h"
+#include "peer-manager.h"
+
+typedef struct {
+        sd_event_source *sd_event_source;
+
+        PeerManager *peer_manager;
+} ConnectionServer;
```

Comment:
I am learning D-Bus, so I'm not too opinionated yet about our use of it.  But I find the name of this type confusing.  Does it serve up D-Bus connections?  And to whom would it serve them?

Reply:
I agree. I think something like ConnectionManager would be better.

Reply:
Initially, I had two objectives in mind:
- separating the creation of the TCP socket and accepting incoming connections from creating the server-side peer dbus (which is done by the `PeerManager`)
- having a struct to control the live-cycle of the `accept_fd`, `event_source` and `PeerManager`

However, looking at it now, it seems that `ConnectionServer` struct is not really needed. The library takes care of closing `accept_fd` by calling `sd_event_source_set_io_fd_own(event_source, true)` and the `PeerManager` is passed in as parameter. I'll have another look and probably remove the struct.